lib/components/album_image.dart

```diff
@@ -219,3 +219,75 @@ class _AlbumImageErrorPlaceholder extends StatelessWidget {
     );
   }
 }
+
+class TapToZoomImage extends StatefulWidget {
+  final AlbumImage albumImg;
+  final bool isDisabled;
+
+  const TapToZoomImage({
+    Key? key,
+    required this.albumImg,
+    this.isDisabled = false,
+  }) : super(key: key);
+
+  @override
+  _TapToZoomImageState createState() => _TapToZoomImageState();
+}
+
+class _TapToZoomImageState extends State<TapToZoomImage> {
+  @override
+  Widget build(BuildContext context) {
+    if (widget.isDisabled) {
+      return widget.albumImg;
+    }
+
+    return GestureDetector(
+      onTap: () => Navigator.of(context).push(
+        PageRouteBuilder(
+          opaque: false,
+          barrierDismissible: true,
```

Comment:
The dismiss gesture works, but it takes half a second for the original cover to re-appear. Not hiding the original cover could help to mitigate that, or if that isn't an option maybe you can play with the transition durations?